packages/hash/api/src/model/ontology/propertyType.model.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  PropertyType as PropertyTypeSchema,
+  GraphApi,
+} from "@hashintel/hash-graph-client";
+
+import { PropertyType as PropertyTypeModel } from "../index";
+
+type PropertyTypeArgs = {
+  accountId: string;
+  schema: PropertyTypeSchema;
+};
+
+class __PropertyType {
+  accountId: string;
+
+  schema: PropertyTypeSchema;
+
+  constructor({ schema, accountId }: PropertyTypeArgs) {
+    this.accountId = accountId;
+    this.schema = schema;
+  }
+
+  /**
+   * Create a property type.
+   *
+   * @param params.accountId the accountId of the account creating the property type
+   * @param params.schema a `PropertyType`
+   */
+  static async create(
+    graphApi: GraphApi,
+    params: {
+      accountId: string;
+      schema: PropertyTypeSchema;
+    },
+  ): Promise<PropertyTypeModel> {
+    const { data: schema } = await graphApi.createPropertyType(params);
+
+    return new PropertyTypeModel({ schema, accountId: params.accountId });
+  }
+
+  /**
+   * Get latest versions of all property types.
+   *
+   * @param params.accountId the accountId of the account requesting the property types
+   */
+  static async getAllLatest(
+    graphApi: GraphApi,
+    params: { accountId: string },
+  ): Promise<PropertyTypeModel[]> {
+    /** @todo: get all latest property types in specified account */
+    const { data: schemas } = await graphApi.getLatestPropertyTypes();
+
+    return schemas.map(
+      (schema) =>
+        new PropertyTypeModel({ schema, accountId: params.accountId }),
+    );
+  }
```

Comment:
@Alfred-Mountfield @TimDiekmann currently we don't have a way of filtering responses from the Graph API by their `accountId`, because the `accountId` isn't returned in the response of any Graph API methods for types - only the schema is.

Are we planning on introducing something like the "qualified type" in the Graph API so that type metadata that isn't captured in the schema can be returned in the API methods?

Reply:
Gotcha, it'd be good to be able to consistently access things like `accountId`, `createdAt`, etc across types and data.